src/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.cs

```diff
@@ -152,39 +154,56 @@ public partial class Regex : System.Runtime.Serialization.ISerializable
         public System.Text.RegularExpressions.RegexOptions Options { get { throw null; } }
         public bool RightToLeft { get { throw null; } }
         public static string Escape(string str) { throw null; }
+        public static int Escape(ReadOnlySpan<char> str, Span<char> destination) { throw null; }
         public string[] GetGroupNames() { throw null; }
         public int[] GetGroupNumbers() { throw null; }
         public string GroupNameFromNumber(int i) { throw null; }
         public int GroupNumberFromName(string name) { throw null; }
         protected void InitializeReferences() { }
         public bool IsMatch(string input) { throw null; }
+        public bool IsMatch(ReadOnlySpan<char> input) { throw null; }
         public bool IsMatch(string input, int startat) { throw null; }
+        public bool IsMatch(ReadOnlySpan<char> input, int startat) { throw null; }
```

Comment:
Overloads that take additional index are not necessary with Span. The caller can slice the Span instead.

Reply:
@jkotas the index here called `startat` does not mark the start of the string but the location where to start searching. This has different semantics as you can start searching at position 6 but the original input string will still be the full string. I think this isn't relevant for Regex.IsMatch but for at least Regex.Split and Regex.Replace and probably also for Regex.Match.

Reply:
For things like `Regex.Replace`, this `startat` means copy everything up to `startat` to the destination Span, and then run regular Replace that does not take `startat` method. So this looks like a convenience method to me - it saves you from typing a tiny bit of code in rare cases to achieve the same effect.

A good question for API review discussion whether this convenience method is worth having.

Reply:
What you describe also applies to at least Regex.Split. I need to look into the other calls if there any differences. Thanks for the input, I will add it to the points to discuss during API review.